src/libraries/System.Private.CoreLib/src/System/Char.cs

```diff
@@ -81,7 +81,8 @@ namespace System
         };
 
         // Return true for all characters below or equal U+00ff, which is ASCII + Latin-1 Supplement.
-        private static bool IsLatin1(char c) => (uint)c < (uint)Latin1CharInfo.Length;
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
Do you have an example where it's not inlined currently?

casts are indeed not needed (since net8)

Reply:
Thanks, I see it's not inlined, but as far as I remember I already complained about it somewhere, the problem was that RVA spans emit an IL that JIT's inliner is not able analyze properly (it doesn't know that it's foldable)

![image](https://user-images.githubusercontent.com/523221/230374773-733805d3-c379-45ee-85dd-9d1d54015b31.png)

Reply:
@EgorBo Sorry for not being clear, since `Char` is a struct the method is in fact inlined ([sharplab.io](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgCUBXAOwwEt8YLAMIR8AB14AbGFADKMgG68wMXAG4aNYgGYGsbABMIXSQE8GuDFA5gMDIQwDeNBq4ZiovBdgwwGkLksGYFNfBgBJXAB1AAteX1kxbBUAMUlsAHMGAF4GOgQADjoNajcGFzcdYIgISQjouISklQAKMBjsKH8AShyAPgqy10iAGR9eLnI23oAyGYYWsb5JoQ6ocK4AMwgAbTAAXQZ5/KLegEJc4sHy0rcAejudgFkYDBiIA3DxSRaXt4+vmJJAB5MR8Yy4FgAQQyGVguFwXhgG0kEwmGW6+2uHi8Pj8zCQ1Vq9SWEym7U6PX6/gYAB4GKSVmsNtsWCMYFwMm8SmVsZ5vGECQw2DBDMCTKZEtguLSQr4+gzxkzOiyINSuDAAO7BUIwHZY26uZyGhgPVyMMrkIakIa6MooIYAViGhLKAHYhgUhgBOIZQoYAISGDjKABEhgBRIYpa5lfJ0CNoPIIBNJ+OJ5OpzMZ9NplM5/N5rMnAsFUvlvNlosFrNmgCqAGo6M26CwWI2WzGTXHC9nq/2+4Pc0Pe8Ox6OJ8XJxn603m6x23PyF2hm4TgG8+QCpvt8mt5uoTuj3vd/lyOg947j2fvdeEORtHf96a7gwO83SG336QV6vM6eUwAuggJAotQP/MDIJPJ9bz3WCb2PWcW20L8520X9VzPACUDoPMcLw3Dk3wojCPyYiyNIhByKoyjqLo2iGIIpMkObFBUNYjChgopiSJ47jeIE/ihJovj7wvLDNyvPcNz3UhmNfd86EddilM4ntyBk/IKDzbTk10rTyB0wy9OMgyjPMkyLLMyybOsli6CQFSkDUtcEH0tzTI8qyvNsnzrP8vzAvPTd4PvKSEMHey3RUt0XPNacB3HKcqxHKc0prDLEoSyKFLnYCVIKOLUsy4qsvSsqSqSyrsvs70VO9Ir1yfQ89xas82vvDryCEJ9NPvHrM3E+8kBCosUk3Aazxk+yoRUqFGv60bMw6ugur69znwk6SixWobyDdHbDuWxDcpbAMVIDBb6NE67BJEu7buEx77qewj7KEFShCuxiHp+16lv+36br+nz7NDFTQwW9zoc8mHvLh3yEYCpHAt0+yIxUiModhnH4dx3zyFClHifx5HjPslIVJSTiAF8ShpoA)).

Reply:
@xtqqczze change host type from struct to class

Reply:
Filed https://github.com/dotnet/runtime/issues/84416

Reply:
>  Is it worth keeping the AggressiveInlining to express intent, or not?

No, but thanks